### DISCUSSION TITLE
Same fontname and encoding but different symbolic name

### DISCUSSION THREAD

**Discussion by meghanaviyyapu** [pymupdf/PyMuPDF#3738] [Looking for help]
In the PDF tagged1.pdf, when I execute page.get_fonts() there are multiple font properties with the same name "FrutigerboldNormal" and MacRomanEncoding but the symbolic names of fonts are different. In this case, how to identify which symbolic name works and which doesn't while inserting text using page.insert_text()?

Attached the output of page.get_fonts(), 
1.pdf  in which the text "CR200078" was inserted using symbolic name T1_0  and it worked fine
3.pdf  in which the text was inserted using T1_1 symbolic name. There was no text but some weird character

Could you let me know why some symbolic names from page.get_fonts() are able to insert text well and why some are not? How to identify the correct one?
[1.pdf](https://github.com/user-attachments/files/16410684/1.pdf)
[tagged1.pdf](https://github.com/user-attachments/files/16410687/tagged1.pdf)
[3.pdf](https://github.com/user-attachments/files/16410688/3.pdf)
![Pagefonts](https://github.com/user-attachments/assets/051178d4-8a5f-4e1d-a36e-34eebbf95c36)

**Comment by JorjMcKie** [comment]
There is no way to systematically find this out.
At least the following code does work:
```python
import pymupdf

doc = pymupdf.open("3.pdf")
page = doc[0]
pnt = (100, 100)
fontbuffer = doc.extract_font(555)[-1]
page.insert_font(fontname="f0", fontbuffer=fontbuffer)
text = "CR200078"
page.insert_text(pnt, text, color=(1, 0, 0), fontname="f0")

doc.ez_save("x.pdf")
``

**Comment by meghanaviyyapu** [comment]
Thanks for the explanation
